ark-blockstore-example/noop-plugin.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/heptio/ark/pkg/cloudprovider"
+	"github.com/heptio/ark/pkg/plugin"
+	"github.com/heptio/ark/pkg/util/collections"
+	"github.com/sirupsen/logrus"
+	"math/rand"
+	"strconv"
+	// TODO: Ark plugin needs to import types from
+	// the kubernetes repo ? What is the guidance about
+	// versioning ?
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Volume keeps track of volumes created by this plugin
+type Volume struct {
+	volType, az string
+	iops        int64
+}
+
+// Snapshot keeps track of snapshots created by this plugin
+type Snapshot struct {
+	volID, az string
+	tags         map[string]string
+}
+
+// Plugin for containing state for the blockstore plugin
+type NoOpBlockStore struct {
+	config    map[string]string
+	logger    logrus.FieldLogger
+	volumes   map[string]Volume
+	snapshots map[string]Snapshot
+}
+
+var _ cloudprovider.BlockStore = (*NoOpBlockStore)(nil)
+
+func NewNoOpBlockStore() *NoOpBlockStore {
+	return &NoOpBlockStore{logger: plugin.NewLogger()}
+}
+
+func (p *NoOpBlockStore) log(msg string, args ...interface{}) {
```

Comment:
you might want to consider embedding the `logrus.Logger` in `NoOpBlockStore` instead of having this wrapper.  The downside to the wrapper is that all log statements will say they came from L45 instead of the place where your helper function is called.